README.md

```diff
@@ -9,92 +9,388 @@
 
 ## Overview
 
-We can now get students thinking outside of the component box and in terms of
-somewhat larger applications. We can move from simply having handlers outside
-of our components to keeping state outside of components as well. We're not
-quite ready for full-blown flux, but we at least have the concepts of actions
-and stores, and we can already make use of them:
+In this lesson we're going to learn about a concept called "unidirectional"
+data-flow. Web Applications are primarily centered around rendering application
+state and data. It's extremely important to choose an architecture that allows
+us to scale our application as we add new components and functionality. Thinking
+about how our components, actions and state containers communicate is crucial
+in order to organize everything beyond trivial demos.
 
-```javascript
-// stores/Store.js
-module.exports = class Store {
-  constructor(state) {
-    this.listeners = []
-    this.state = state
+## Towards Unidirectional Data Flow
+
+### Using Change Listeners
+
+To understand what "unidirectional data flow" actually means, let's first
+consider a real-world example:
```

Comment:
I think it'd be better for readability if we kept this sentence and the next one in one paragraph, as well as end the first sentence with a period.